### PR TITLE
Add initial rest interface

### PR DESCRIPTION
**Pull request by sashjakk**

### CODE REVIEW

File: src/main/scala/com/github/sashjakk/http/Codecs.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    interval => Json.obj(("from", interval.from.asJson), ("to", interval.to.asJson))
+  implicit val intervalDecoder: Decoder[Interval] =
+    cursor =>
+      for {
+        from <- cursor.downField("from").as[Instant]
+        to <- cursor.downField("to").as[Instant]
+        interval <- {
+          type ErrOr[A] = Either[Throwable, A]
+
+          Interval.from[ErrOr](from, to) match {
```
Comment (sashjakk) [inline src/main/scala/com/github/sashjakk/http/Codecs.scala:36]: Is it possible to use `Decoder.Result` directly? Cause it's also typealias for Either

Comment (kurnevsky) [reply src/main/scala/com/github/sashjakk/http/Codecs.scala:36]: Yes, you can use it for everything that implements MonadThrow (you actually need only ApplicativeThrow). That's the point of tagless final :)

File: src/main/scala/com/github/sashjakk/http/Server.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        .redeemWith(e => BadRequest(APIError(e.getMessage)), Created(_))
+    }
+  }
+
+  private def spots(service: SpotService[IO]): HttpRoutes[IO] = {
+    HttpRoutes.of[IO] {
+      case GET -> Root / "spots" =>
+        service
+          .sharedSpots()
+          .redeemWith(e => BadRequest(APIError(e.getMessage)), Ok(_))
```
Comment (kurnevsky) [inline src/main/scala/com/github/sashjakk/http/Server.scala]: You can use a [middleware](https://http4s.org/v0.23/docs/middleware.html) to handle this commonly.

File: src/main/scala/com/github/sashjakk/http/Codecs.scala (hunk 1 above)
Comment (sashjakk) [reply src/main/scala/com/github/sashjakk/http/Codecs.scala:36]: not solved 🙈 will continue like this for now, todo left

**Pull request merged**